src/App/DocumentObject.cpp

```diff
@@ -1416,6 +1416,19 @@ bool DocumentObject::redirectSubName(std::ostringstream &, DocumentObject *, Doc
     return false;
 }
 
+Base::Placement DocumentObject::globalPlacement() const {

```

Comment:
This is a code smell because a base class should not make assumptions about a sub-class.

Reply:
I assume its because I use the Placement in that method? I changed the PR and it now includes the `globalGroupPlacement()` you suggested in the comment from a few days ago, assuming that is not a code smell.

Reply:
For some explanations read this: https://luzkan.github.io/smells/base-class-depends-on-subclass
A further problem would be that whenever using `globalPlacement()` it will never be clear what the returned placement will be and thus is suspicious to also break the Liskov substitution principle.

Reply:
Thanks, nice link. I guess this is resolved then.